encoding/encoding_test.go

```diff
@@ -904,3 +904,33 @@ func generateFixedLenByteArrayValues(n int, r *rand.Rand, size int) []byte {
 	io.ReadFull(r, values)
 	return values
 }
+
+func maxLenInt8(data []int8) int {
+	max := 0
+	for _, v := range data {
+		if n := bits.Len8(uint8(v)); n > max {
+			max = n
+		}
+	}
+	return max
+}
+
+func maxLenInt32(data []int32) int {
+	max := 0
+	for _, v := range data {
+		if n := bits.Len32(uint32(v)); n > max {
+			max = n
+		}
+	}
+	return max
+}
+
+func maxLenInt64(data []int64) int {
+	max := 0
+	for _, v := range data {
+		if n := bits.Len64(uint64(v)); n > max {
+			max = n
+		}
+	}
+	return max
+}
```

Comment:
These functions were in the `internal/bits` package but only used here (with one exception), it seemed better to bring them here instead of having a dependency.